phosphor-auditlog/auditLogParser.hpp

```diff
@@ -0,0 +1,167 @@
+    }
+
+    virtual void doParse();
+    void processEvents();
+    void fillAuditEntry(nlohmann::json& parsedEntry);
+    void fillUsysEntry(nlohmann::json& parsedEntry);
+    virtual bool formatEntry(nlohmann::json& parsedEntry)
+    {
+        return formatRaw(parsedEntry);
+    };
+    bool formatMsgReg(nlohmann::json& parsedEntry);
+    bool formatGen(nlohmann::json& parsedEntry);
+    bool formatRaw(nlohmann::json& parsedEntry);
+
+    auparse_state_t* au = nullptr;
+    std::ofstream parsedStream;
+
+  private:
+    bool auditNextEvent();
+    void parseEvent();
+    virtual void parseRecord();
+    bool openParsedFile(std::string filePath);
+};
+
+/** @class ALParseLatest
+ *  @brief Parsing audit log using auparse library services
+ *  @details Provides means to parse only latest entries
+ */
+class ALParseLatest : ALParser
+{
+  public:
+    ALParseLatest(const ALParseLatest&) = delete;
+    ALParseLatest& operator=(const ALParseLatest&) = delete;
+    ALParseLatest(ALParseLatest&&) = delete;
+    ALParseLatest& operator=(ALParseLatest&&) = delete;
+
+    /** @brief Constructor to initialize parsing of audit log files
+     */
+    ALParseLatest(uint32_t maxEvents, ALParseFile& parsedFile) :
+        ALParser(parsedFile)
+    {
+        lg2::debug("Constructing ALParseLatest : max = {MAXCOUNT}", "MAXCOUNT",
+                   maxEvents);
+
+        // Don't allow request without non-zero count
+        if (maxEvents > 0)
+        {
+            initNextLog();
+            maxCount = maxEvents;
+            maxLeftCount = maxEvents;
+        }
+
+        if (au == nullptr)
+        {
+            lg2::error("Failed to init auparse");
+            throw sdbusplus::xyz::openbmc_project::Common::Error::
+                InternalFailure();
+        }
+    }
+
+    void doParse() override;
+    bool formatEntry(nlohmann::json& parsedEntry) override
+    {
+        return formatMsgReg(parsedEntry);
+    };
+
+  private:
+    unsigned int maxCount = 0;
+    unsigned int maxLeftCount = 0;
+    unsigned int logFileIdx = 0;
+    std::list<std::string> parsedEntries;
+
+    void parseRecord() override;
+    bool initNextLog();
+    size_t writeParsedEntries();
+};
+
+/** @class ALParseAll
+ *  @brief Parsing audit log using auparse library services
+ *  @details Provides means to parse all entries
+ */
+class ALParseAll : ALParser
+{
+  public:
+    ALParseAll(const ALParseAll&) = delete;
+    ALParseAll& operator=(const ALParseAll&) = delete;
+    ALParseAll(ALParseAll&&) = delete;
+    ALParseAll& operator=(ALParseAll&&) = delete;
+
+    ALParseAll(ALParseFile& parsedFile) : ALParser(parsedFile)
+    {
+        lg2::debug("Constructing ALParseAll");
+        au = auparse_init(AUSOURCE_LOGS, nullptr);
+        if (au == nullptr)
+        {
+            lg2::error("Failed to init auparse");
+            throw sdbusplus::xyz::openbmc_project::Common::Error::
+                InternalFailure();
+        }
+    }
+
+    void doParse() override
```

Comment:
since this just calls the base class doParse() I don't think you even need the override?